core/src/main/java/hudson/search/Search.java

```diff
@@ -157,10 +160,23 @@ public void doSuggestOpenSearch(StaplerRequest2 req, StaplerResponse2 rsp, @Quer
      */
     public void doSuggest(StaplerRequest2 req, StaplerResponse2 rsp, @QueryParameter String query) throws IOException, ServletException {
         Result r = new Result();
-        for (SuggestedItem item : getSuggestions(req, query))
-            r.suggestions.add(new Item(item.getPath(), item.getUrl()));
+        for (SuggestedItem curItem : getSuggestions(req, query)) {
+            String iconName = curItem.item.getSearchIcon();
```

Comment:
One pre-existing issue is that items with a display name are found twice, (even when the actual name is nothing alike)

<img width="738" alt="image" src="https://github.com/user-attachments/assets/59747e13-60ea-4835-aac0-faccb84e0cf4" />

But the display name version isn't an instanceof Item, which means it just gets the default icon.

Ideally it wouldn't report twice or at least it would provide the right item.

See pre-existing screenshot from ci.jenkins.io:
<img width="449" alt="image" src="https://github.com/user-attachments/assets/129247fe-ce9a-4112-abf6-d6df9574de65" />